### PR TITLE
Spec config in storage

### LINKED ISSUE #3430: Support changing forks when verifying signatures on historical blocks
### Description
Support changing forks when verifying signatures on historical blocks.

Details are here: https://github.com/ConsenSys/teku/blob/ee9b3f5d409370f6e0d8c61da8a13d3aef9876d7/storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java#L305-L307

### PR DESCRIPTION
**Pull request by rolfyone**
Once storageService is using the specProvider, we can dynamically query the fork rather than be hard coded to a single fork.

fixes #3430

Signed-off-by: Paul Harris <paul.harris@consensys.net>

## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java [hunk 1]
```diff
@@ -196,13 +207,16 @@ private RocksDbDatabase(
       final RocksDbFinalizedDao finalizedDao,
       final RocksDbEth1Dao eth1Dao,
       final RocksDbProtoArrayDao protoArrayDao,
-      final StateStorageMode stateStorageMode) {
+      final StateStorageMode stateStorageMode,
+      final SpecProvider specProvider) {
+    checkArgument(specProvider != null);
```
Comment (ajsutton) [inline storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java]: nit: 
```suggestion
    checkNotNull(specProvider);
```

File: storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java [hunk 2]
```diff
@@ -304,13 +318,16 @@ boolean batchVerifyHistoricalBlockSignatures(final Collection<SignedBeaconBlock>
...
-    // to have to change the way we retrieve the domain here.
-    Bytes32 domain = get_domain(finalizedState, DOMAIN_BEACON_PROPOSER);
+    final Bytes32 genesisValidatorsRoot = finalizedState.getForkInfo().getGenesisValidatorsRoot();
 
     blocks.forEach(
         signedBlock -> {
-          BeaconBlock block = signedBlock.getMessage();
+          final BeaconBlock block = signedBlock.getMessage();
+          final UInt64 epoch = compute_epoch_at_slot(block.getSlot());
+          final Fork fork = specProvider.fork(epoch);
```
Comment (ajsutton) [inline storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java:328]: I wonder if it's worth having a `forkAtSlot` method that does the conversion to epoch for you.

**Review by ajsutton — APPROVED** [review]
LGTM.

File: storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java (hunk 2 above)
Comment (rolfyone) [reply storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java:328]: as discussed. need epoch in this instance, but potentially use cases down the line. will leave it for now.

**Pull request merged**